app/src/lib/git/config.ts

```diff
@@ -128,3 +125,104 @@ export async function getGlobalConfigPath(env?: {
 
   return normalize(path[1])
 }
+
+/** Set the local config value by name. */
+export async function setConfigValue(
+  repository: Repository,
+  name: string,
+  value: string,
+  env?: {
+    HOME: string
+  }
+): Promise<void> {
+  setConfigValueInPath(name, value, repository.path, env)
+}
+
+/** Set the global config value by name. */
+export async function setGlobalConfigValue(
+  name: string,
+  value: string,
+  env?: {
+    HOME: string
+  }
+): Promise<void> {
+  setConfigValueInPath(name, value, null, env)
+}
+
+/**
+ * Set config value by name
+ *
+ * @param path The path to execute the `git` command in. If null
+ *             we'll use the global configuration (i.e. --global)
+ *             and execute the Git call from the same location that
+ *             GitHub Desktop is installed in.
+ */
+async function setConfigValueInPath(
+  name: string,
+  value: string,
+  path: string | null,
+  env?: {
+    HOME: string
+  }
+): Promise<void> {
+  const options = env ? { env } : undefined
+
+  const flags = ['config']
+
+  if (!path) {
+    flags.push('--global')
+  }
+
+  flags.push('--replace-all', name, value)
+
+  await git(flags, path || __dirname, 'setConfigValueInPath', options)
+}
+
+/** Remove the local config value by name. */
+export async function removeConfigValue(
```

Comment:
To save you jumping through the code, this is used when the user switches from local user info (name and email) to global user info, to delete the local user info.